### ISSUE TITLE
Removal of Homebridge dependency

### ISSUE THREAD

**Issue by theimo1221** [hjdhjd/unifi-protect#49]
**Describe The Problem:**
Missing module `homebridge-plugin-utils`
````
node_modules/unifi-protect/dist/index.d.ts:8:26 - error TS2307: Cannot find module 'homebridge-plugin-utils' or its corresponding type declarations.

8 export { Nullable } from "homebridge-plugin-utils";
````

**To Reproduce:**

I tried using this in https://github.com/Hoffmation/Hoffmation-Base which uses Typescript as well.

**Comment by theimo1221** [comment]
Shall I open a PR moving the utils module to normal dependencies?

**Comment by theimo1221** [comment]
I did some further inspecting.
Adding `homebridge-plugin-utils` I'd need `homebridge` too:
```
node_modules/homebridge-plugin-utils/dist/service.d.ts:1:59 - error TS2307: Cannot find module 'homebridge' or its corresponding type declarations.

1 import { HAP, PlatformAccessory, Service, WithUUID } from "homebridge";
```
So I'd highly appreciate if we could separate this from homebridge, as if I understand it correctly thats the main purpose of `unifi-protect` vs https://github.com/hjdhjd/homebridge-unifi-protect

**Comment by theimo1221** [comment]
Once this is complete I can do it in unifi-access as well🙂

**Comment by hjdhjd** [comment]
Dependencies are a fact of life in the node package world. 😄 I have no plans to remove the dependency on my utility library - it explicitly exists to **reduce** duplication across a variety of things I do, and will continue to build into my libraries in the future.

TL;DR: this behavior is by design and not going anywhere.

**Comment by theimo1221** [comment]
Thanks for the feedback. In my opinion I'm good with programming/logic but rather bad in reporting issues or proposing changes to others and their "baby" without hurting them.

Neither do I want to diminish your work nor the time/effort to maintain/explore all those proprietary information/interfaces. 

Regarding `The issue you raised about exporting a type declared as a development dependency will be addressed in a future release.` can I lend you a hand with this?

While my 3rd-Person view might be shortsighted as I'd never be as deep/involved in it as you are I'm still certain the current overlap/entangling of these separate packages might result in plenty of headache in the future. 
Just to quote your https://github.com/hjdhjd/homebridge-plugin-utils readme:
> homebridge-plugin-utils is a utility library for [Homebridge](https://homebridge.io/) [plugins](https://developers.homebridge.io/) that aims to provide a set of common core capabilities that can accelerate and streamline plugin development

So as stated by yourself it is a utility library for Homebridge plugins and their development and if I understand this package and your intentions with it correct this should be the independent base of https://github.com/hjdhjd/homebridge-unifi-protect so it's not an homebridge plugin.

Best Regards

Thiemo
